vendor/k8s.io/kubernetes/pkg/util/iptables/iptables.go

```diff
@@ -323,6 +326,27 @@ func (runner *runner) RestoreAll(data []byte, flush FlushFlag, counters RestoreC
 	return runner.restoreInternal(args, data, flush, counters)
 }
 
+// Basically duplicate iptables 1.6.x xtables_lock() function.  We don't bother
+// trying to acquire the 1.4.x-style lock socket since both F25 and RHEL7
+// use the lockfile approach.
+func grabIptablesLock() (*os.File, error) {
+	// First acquire the iptables 1.6.x lockfile (eg, Fedora 25+ and RHEL7)
+	fd, err := os.OpenFile("/run/xtables.lock", os.O_CREATE, 0600)
+	if err != nil {
+		// Ignore errors opening the lockfile like xtables_lock() does
+		return nil, nil
+	}
+
+	// Try to grab it for 5 seconds before giving up
+	for i := 0; i < 5; i++ {
```

Comment:
@danwinship would you prefer to just use 2 seconds to correspond to the existing "-w2" (eg, wait for 2 seconds for the lock then fail) argument for iptables?